### ISSUE TITLE
Allow usernames to contain any special character

### ISSUE THREAD

**Issue by PascaleBeier** [pagekit/pagekit#713]
Hey,

Why don't you allow usernames to contain any special character? E.g. my client's usernames are their first- and lastname and they are really used to it. So am I 😄 

From a security perspective it doesn't make too much sense restricting username input to these few special characters.

I would love submitting a pull request if this is something we could change.

**Comment by MalteScharenberg** [comment]
The username is just for the login. At your profile the name gets displayed (which supports spaces).

**Comment by PascaleBeier** [comment]
I'm really talking about the username for login. I find it very limiting and concerning in terms of security.

Which CMS or Framework you know of doesn't allow you to use special chars in a username? Even WordPress does - and you really should know that 🙏

**Comment by MalteScharenberg** [comment]
I agree, I can't see any argument against allowing special characters as login. But on the contrary it even limits the possible combinations of logins names, so that one could argue that is could facilitate brute force attacks.  
I will investigate why it was implemented only permitting a very limited range of special chars.

**Comment by jdegger** [comment]
How would it facilitate a brute force attack if there are more possible combinations of characters to choose from?

I mean I understand you don't want two almost the same usernames but that shouldn't validate the login. Example:

Username: Jésse

Makes it impossible to register with:
- Jésse
- Jesse
- Jessé
- Jéssé
- etc.

Only lets you login wit the exact username Jésse.

**Comment by PascaleBeier** [comment]
You are kind of missing the point as that is exactly what I was saying :)

Am 26.07.2016 19:37 schrieb "Jesse Degger" <notifications@github.com>:

**Comment by MalteScharenberg** [comment]
@jdegger I think I have expressed myself unclear.  I meant the exact same thing you are describing. 😊
